lib/adal.js

```diff
@@ -494,6 +495,113 @@ AuthenticationContext.prototype.acquireToken = function (resource, callback) {
+    if (this._isEmpty(resource)) {
+      this.warn('resource is required');
+      callback('resource is required', null, 'resource is required');
+      return;
+    }
+  
+    var callback = this.callback;
+  
+    if (!this._user) {
+      this.warn('User login is required');
+      callback('User login is required', null, 'login required');
+      return;
+    }
+  
+    if (this._acquireTokenInProgress) {
+      this.warn("Acquire token interactive is already in progress")
+      callback("Acquire token interactive is already in progress", null, "Acquire token interactive is already in progress");
+      return;
+    }
+  
+    var expectedState = this._guid() + '|' + resource;
+    this.config.state = expectedState;
+    this.verbose('Renew token Expected state: ' + expectedState);
+  
+    // remove the existing prompt=... query parameter and add prompt=select_account
+    var urlNavigate = this._urlRemoveQueryStringParameter(this._getNavigateUrl('token', resource), 'prompt');
+    urlNavigate = urlNavigate + '&prompt=select_account';
+    if (extraQueryParameters) {
+      urlNavigate += extraQueryParameters;
+    }
+  
+    if (claims && (urlNavigate.indexOf("&claims") === -1)) {
+      urlNavigate += '&claims=' + encodeURIComponent(claims);
+    }
+    else if (claims && (urlNavigate.indexOf("&claims") !== -1)) {
+      throw new Error('Claims cannot be passed as an extraQueryParameter');
+    }
+  
+    urlNavigate = this._addHintParameters(urlNavigate);
+    this._acquireTokenInProgress = true;
+    this.info('acquireToken interactive is called for the resource ' + resource);
+    this._saveItem(this.CONSTANTS.STORAGE.LOGIN_REQUEST, window.location.href);
+    this._saveItem(this.CONSTANTS.STORAGE.STATE_RENEW, expectedState, true);
+    this.promptUser(urlNavigate);
+  };
+  
+  /**
+   * Removes the query string parameter from the authorization endpoint URL if it exists
+   * @ignore
+   */
+  AuthenticationContext.prototype._urlRemoveQueryStringParameter = function (url, name) {
+    // we remove &name=value, name=value& and name=value
+    // &name=value
+    var regex = new RegExp('(\\&' + name + '=)[^\&]+');
+    url = url.replace(regex, '');
+    // name=value&
+    regex = new RegExp('(' + name + '=)[^\&]+&');
+    url = url.replace(regex, '');
+    // name=value
+    regex = new RegExp('(' + name + '=)[^\&]+');
+    url = url.replace(regex, '');
+    return url;
+  }
+  
+  /**
+   * Adds login_hint to authorization URL which is used to pre-fill the username field of sign in page for the user if known ahead of time.
+   * domain_hint can be one of users/organisations which when added skips the email based discovery process of the user.
+   * @ignore
+   */
+  AuthenticationContext.prototype._addHintParameters = function (urlNavigate) {
+    //If you dont use prompt=none, then if the session does not exist, there will be a failure.
```

Comment:
there's an unknown character in dont

Reply:
Fix this, it's not shown up on my VS code, but it has the special character when showing in github desktop client.